analytics_data_api/v0/connections.py

```diff
@@ -0,0 +1,69 @@
+import json
+import time
+
+from boto.connection import AWSAuthConnection
+from django.conf import settings
+from elasticsearch import Connection
+
+
+class BotoHttpConnection(Connection):
+    """
+    Uses AWS configured connection to sign requests before they're sent to
+    elasticsearch nodes.
+    """
+
+    connection = None
+
+    def __init__(self, host='localhost', port=443, **kwargs):
+        super(BotoHttpConnection, self).__init__(host=host, port=port, **kwargs)
+        connection_params = {'host': host, 'port': port}
+
+        # Set the credentials through settings.  If not provided, boto will attempt to use
+        # default environment variables.
```

Comment:
I was hoping that you could pass arbitrary keyword arguments into create_connection and they would be passed along to the Connection class when it actually *is* instantiated.

Something like:

```python
connection_params = {'hosts': [settings.ELASTICSEARCH_LEARNERS_HOST]}
if settings.ELASTICSEARCH_CONNECTION_CLASS:
           connection_params['connection_class'] = class_for_name(settings.ELASTICSEARCH_CONNECTION_CLASS)
if settings.ELASTICSEARCH_AWS_ACCESS_KEY_ID:
           connection_params['aws_access_key_id'] = settings.ELASTICSEARCH_AWS_ACCESS_KEY_ID
if settings.ELASTICSEARCH_AWS_SECRET_ACCESS_KEY:
          connection_params['aws_secret_access_key'] = settings.ELASTICSEARCH_AWS_SECRET_ACCESS_KEY
           connections.connections.create_connection(**connection_params)
```

If that doesn't work, I'm OK with just wrapping the import in a try-except block that catches the ImportError.